File: src/main/java/com/indiepost/repository/AdminPostRepository.java
package com.indiepost.repository;

import com.indiepost.dto.post.AdminPostSummaryDto;
import com.indiepost.dto.post.PostQuery;
import com.indiepost.dto.post.Title;
import com.indiepost.enums.Types;
import com.indiepost.model.Post;
import com.indiepost.model.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * Created by jake on 17. 1. 10.
 */
public interface AdminPostRepository {

    Long persist(Post post);

    Post findOne(Long id);

    void delete(Post post);

    void deleteById(Long id);

    boolean isExists(Long id);

    List<AdminPostSummaryDto> findByIdIn(List<Long> id);

    List<AdminPostSummaryDto> find(User currentUser, Pageable pageable);

    List<AdminPostSummaryDto> find(User currentUser, Types.PostStatus status, Pageable pageable);

    List<Post> findAll();

    List<Post> findByIds(List<Long> ids);

    Page<AdminPostSummaryDto> findText(String text, User currentUser, Types.PostStatus status, Pageable pageable);

    List<Long> findIds(User currentUser, Types.PostStatus status);

    List<String> findAllDisplayNames();

    List<Title> getTitleList();

    Long count();

    Long count(PostQuery postQuery);

    Long count(Types.PostStatus status, User currentUser);

    List<Post> findScheduledToBePublished();

    void disableSplashPosts();

    void disableFeaturedPosts();
}

File: src/main/kotlin/com/indiepost/repository/AdminPostRepository.kt
package com.indiepost.repository

import com.indiepost.dto.post.AdminPostSummaryDto
import com.indiepost.dto.post.Content
import com.indiepost.dto.post.PostQuery
import com.indiepost.dto.post.Title
import com.indiepost.enums.Types
import com.indiepost.model.Post
import com.indiepost.model.User
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Pageable

/**
 * Created by jake on 17. 1. 10.
 */
interface AdminPostRepository {

    fun getTitleList(): List<Title>

    fun persist(post: Post): Long?

    fun findOne(id: Long?): Post?

    fun delete(post: Post)

    fun deleteById(id: Long?)

    fun isExists(id: Long?): Boolean

    fun findByIdIn(ids: List<Long>): List<AdminPostSummaryDto>

    fun find(currentUser: User, pageable: Pageable): List<AdminPostSummaryDto>

    fun find(currentUser: User,
             status: Types.PostStatus,
             tag: String? = null,
             pageable: Pageable = PageRequest.of(0, 50)): List<AdminPostSummaryDto>

    fun findAll(): List<Post>

    fun findByIds(ids: List<Long>): List<Post>

    fun findText(text: String, currentUser: User, status: Types.PostStatus, pageable: Pageable): Page<AdminPostSummaryDto>

    fun findIds(currentUser: User, status: Types.PostStatus): List<Long>

    fun findAllDisplayNames(): List<String>

    fun count(): Long

    fun count(postQuery: PostQuery): Long

    fun count(status: Types.PostStatus, currentUser: User, tag: String? = null): Long

    fun findScheduledToBePublished(): List<Post>

    fun disableSplashPosts()

    fun disableFeaturedPosts()

    fun findContentListIncludingVideo(): List<Content>

    fun findBrokenPosts(pageable: Pageable): List<AdminPostSummaryDto>

    fun countBrokenPosts(): Long
}
